src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -3051,56 +3051,74 @@ def get_rotation_poll_interval(self) -> Union[str, None]:
     # pylint: disable=unused-argument
     def _get_enable_aad(self, enable_validation: bool = False, **kwargs) -> bool:
         """Internal function to obtain the value of enable_aad.
-        This function supports the option of enable_validation. When enabled, if the value of enable_aad is True and
-        any of aad_client_app_id, aad_server_app_id or aad_server_app_secret is asssigned, a
+
+        This function supports the option of enable_validation. When enabled, in create mode, if the value of enable_aad
+        is True and any of aad_client_app_id, aad_server_app_id or aad_server_app_secret is asssigned, a
         MutuallyExclusiveArgumentError will be raised. If the value of enable_aad is False and the value of
-        enable_azure_rbac is True, a RequiredArgumentMissingError will be raised.
+        enable_azure_rbac is True, a RequiredArgumentMissingError will be raised. In update mode, if enable_aad is
+        specified and managed aad has been enabled, an InvalidArgumentValueError will be raised.
+
         :return: bool
         """
         # read the original value passed by the command
         enable_aad = self.raw_param.get("enable_aad")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.aad_profile and
-            self.mc.aad_profile.managed is not None
-        ):
-            enable_aad = self.mc.aad_profile.managed
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
```

Comment:
Could you please describe more about the behavior? Why should we try to read the property value corresponding to the parameter from the `mc` object in **create** mode?

Reply:
Sure, currently we are refactoring `aks create` and `aks update`, and "In create mode" means this only applies to the case of `aks create` but not `aks update`.

Reply:
In the update mode, the parameter value is not obtained from mc because we just need to use the value provided by the user to overwrite the value in mc.

But in the create mode, mc is completely blank, and we want to assemble the user's input into a complete mc. Considering that our parameters have a certain dependency or mutual exclusion relationship, or the parameters will be normalized before putting into mc, there are even some parameters that do not need user to provide explicitly, but can be dynamically completed by cli.

In order to ensure the consistency of the state and make the value of the parameter more clearly determined, we created the following principle in the create mode, that is, if a parameter has been placed in the mc object, then when trying to get the parameter value again, (e.g. cross-validation with other parameters) the value in mc shall prevail.